### PR TITLE
Allocation all code (do not merge - reviewed parts are in "allocation" branch).

### PR DESCRIPTION
**Pull request by ppalucki**
Long-lived branch that includes allocation feature:

TODO:
- [x] refactor runners to extract common functinoallity
- [x] RDT resources allocation
- [x] RDT independent monitoring and allocations (MON_GROUPS cs CTRL_GROUPS)
- [x] cgroup based resources allocation
- [x] extra profiling metrics
- [x] refactor labeling (as metrics package)
- [x] calculating resulting allocations
- [x] encoding allocations as metrics
- [x] unit tests:
  - [x] Cgroup object
  - [x] ResGroup object
  - [x] resulting allocations 
  - [x] AllocationRunner/ContainerManager and Containers
- [x] documentation (another PR)
- [x] simple allocator example
- [x] extend platform information 
- [x] validation of returned allocations (RDT )
- [x] validation of returned allocations (non RDT (shares, quota))
- [x] default RDT allocation and proper regroup cleaning in the initialization phase
- [x] implemented abstract based classed based serialization and merging of allocation value
- [x] functional/e2e tests on dedicated hardware
   - [x] removing and add workloads in the same ResGroups
   - [x] garbage collecting of unused ResGroups

Bugs:
- [x] race condition for invalid pids, while synchronizing cgroups/tasks with resgroup/tasks
- [x] reappearing root/mon_groups

### CODE REVIEW

File: owca/allocators.py [hunk 1]
```diff
@@ -48,13 +90,16 @@ class RDTAllocation:
 class AllocationConfiguration:
 
     # Default value for cpu.cpu_period [ms] (used as denominator).
-    cpu_quota_period: int = 100
+    cpu_quota_period: int = 1000
```
Comment (ss7pro) [inline owca/allocators.py]: Why we're changing this?

File: configs/mesos_example_allocator.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2018 Intel Corporation
```
Comment (bwplotka) [inline configs/mesos_example_allocator.yaml]: Hm.. Year looks wrong to me (:

File: owca/allocators.py (hunk 1 above)
Comment (ppalucki) [reply owca/allocators.py]: because it is the minmal value

File: configs/mesos_example_allocator.yaml (hunk 2 above)
Comment (ppalucki) [reply configs/mesos_example_allocator.yaml]: It was committed to finish, last year :) I don't want to create proof that we are out of schedule ! :innocent:

**Comment by felidadae** [comment]
The branch for the PR is used by PRC team and is frozen.

**Comment by ppalucki** [comment]
super seeded #10

**Pull request closed without merging**